src/modules/idle_inhibitor.cpp

```diff
@@ -19,37 +25,60 @@ waybar::modules::IdleInhibitor::~IdleInhibitor() {
     zwp_idle_inhibitor_v1_destroy(idle_inhibitor_);
     idle_inhibitor_ = nullptr;
   }
+
+  // Remove this from the modules list
+  waybar::modules::IdleInhibitor::modules.remove(this);
+
   if (pid_ != -1) {
     kill(-pid_, 9);
     pid_ = -1;
   }
 }
 
 auto waybar::modules::IdleInhibitor::update() -> void {
+  // Check status
+  std::string status = waybar::modules::IdleInhibitor::status;
+  if (status == "activated") {
+    if (idle_inhibitor_ == nullptr) {
+      idle_inhibitor_ = zwp_idle_inhibit_manager_v1_create_inhibitor(
+        waybar::Client::inst()->idle_inhibit_manager, bar_.surface);
+    }
+  } else {
+    if (idle_inhibitor_ != nullptr) {
+      zwp_idle_inhibitor_v1_destroy(idle_inhibitor_);
+      idle_inhibitor_ = nullptr;
+    }
+  }
+
   label_.set_markup(
-      fmt::format(format_, fmt::arg("status", status_), fmt::arg("icon", getIcon(0, status_))));
-  label_.get_style_context()->add_class(status_);
+      fmt::format(format_, fmt::arg("status", status), fmt::arg("icon", getIcon(0, status))));
+  label_.get_style_context()->add_class(status);
   if (tooltipEnabled()) {
-    label_.set_tooltip_text(status_);
+    label_.set_tooltip_text(status);
   }
   // Call parent update
   ALabel::update();
 }
 
 bool waybar::modules::IdleInhibitor::handleToggle(GdkEventButton* const& e) {
   if (e->button == 1) {
-    label_.get_style_context()->remove_class(status_);
-    if (idle_inhibitor_ != nullptr) {
-      zwp_idle_inhibitor_v1_destroy(idle_inhibitor_);
-      idle_inhibitor_ = nullptr;
-      status_ = "deactivated";
+    std::string status = waybar::modules::IdleInhibitor::status;
```

Comment:
You don't need to use fully qualified name to refer to static class member.
It might be also worth to store status as a `static bool active` since string representations are only used for UI update. I.e. the construction here shortens to `active = !active;`

Reply:
I've pushed a fix, I now just use the static class member directly.

Reply:
Also, I agree about making status a bool, pushed another commit.